assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -35,8 +38,55 @@ module algorithm_info_mod
 integer, parameter :: BOUNDED_NORMAL_RHF = 101 
 
 public :: obs_error_info, probit_dist_info, obs_inc_info, &
+          init_algorithm_info_mod, end_algorithm_info_mod, &
           EAKF, ENKF, BOUNDED_NORMAL_RHF, UNBOUNDED_RHF, GAMMA_FILTER
 
+!Creates the type definitions for the QCF table
+type obs_error_info_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_inflation_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_extended_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type obs_inc_info_type
+   character(len=129)    :: filter_kind
+   logical               :: rectangular_quadrature, gaussian_likelihood_tails
+   logical               :: sort_obs_inc, spread_restoration
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type qcf_table_data_type
+   type(obs_error_info_type)        :: obs_error_info
+   type(probit_inflation_type)      :: probit_inflation
+   type(probit_state_type)          :: probit_state
+   type(probit_extended_state_type) :: probit_extended_state
+   type(obs_inc_info_type)          :: obs_inc_info
+end type
+
+character(len=129), dimension(4) :: header1
+character(len=129), dimension(29) :: header2
+
+character(len=129), allocatable :: qcf_table_row_headers(:)
```

Comment:
I'm finding the name `qcf_table_row_headers`  a bit confusing when reading the code.  It is the bounded quantities, so maybe just call it bounded_qtys:

```diff
-character(len=129), allocatable :: qcf_table_row_headers(:)
+character(len=129), allocatable :: bounded_qtys(:)
```

```diff
-!find location of QTY in qcf_table_data structure
-QTY_loc = findloc(qcf_table_row_headers, qty_name)
+QTY_loc = findloc(bounded_qtys, qty_name)
```

Reply:
> I'm finding the name `qcf_table_row_headers` a bit confusing when reading the code. It is the bounded quantities, so maybe just call it bounded_qtys:
> 
> ```diff
> -character(len=129), allocatable :: qcf_table_row_headers(:)
> +character(len=129), allocatable :: bounded_qtys(:)
> ```
> 
> ```diff
> -!find location of QTY in qcf_table_data structure
> -QTY_loc = findloc(qcf_table_row_headers, qty_name)
> +QTY_loc = findloc(bounded_qtys, qty_name)
> ```

To correct myself here, it is not bounds only. You could be giving a different distribution also, and a different filter kind.

Maybe 
`special_qtys`
`non_normal_qtys` ?

Reply:
@hkershaw-brown what about `specified_qtys`?